frontend/src/metabase/static-viz/components/WaterfallChart/WaterfallChart.jsx

```diff
@@ -0,0 +1,342 @@
+  const topMargin = settings.show_values
+    ? layout.margin.top + VALUES_MARGIN
+    : layout.margin.top;
+
+  const xTickHeight = isVertical ? xTickWidth : chartStyle.axes.ticks.fontSize;
+  const yTickWidth = getYTickWidth(
+    data,
+    accessors,
+    settings,
+    chartStyle.axes.ticks.fontSize,
+  );
+  const yLabelOffset = yTickWidth + layout.labelPadding;
+  const xMin = yLabelOffset + chartStyle.axes.labels.fontSize * 1.5;
+  const xMax = layout.width - layout.margin.right - layout.margin.left;
+  const xAxisHeight = getXAxisHeight(
+    xTickHeight,
+    measureTextHeight(chartStyle.axes.labels.fontSize),
+  );
+  let yMax = layout.height - xAxisHeight - topMargin;
+  let height = layout.height;
+  // If inner chart area is too short, try to expand it but not more than `MAX_EXTRA_HEIGHT`
+  // to match what we do with XYChart (with legends, it can be up to 340px tall)
+  if (yMax < MIN_INNER_HEIGHT) {
+    yMax =
+      minMax(
+        layout.height,
+        layout.height + MAX_EXTRA_HEIGHT,
+        yMax + xAxisHeight + MAX_EXTRA_HEIGHT,
+      ) - xAxisHeight;
+    height = topMargin + yMax + xAxisHeight;
+  }
+  const leftLabel = labels?.left;
+
+  const xScale = scaleBand({
+    domain: entries.map(entry => entry.x),
+    range: [0, xMax],
+    padding: layout.barPadding,
+  });
+
+  const yScale = scaleLinear({
+    domain: calculateWaterfallDomain(entries),
+    range: [yMax, 0],
+  });
+
+  const getBarProps = entry => {
+    const width = xScale.bandwidth();
+
+    const height = Math.abs(yScale(entry.start) - yScale(entry.end));
+    const x = xScale(entry.x);
+    const y = yScale(Math.max(entry.start, entry.end));
+
+    const fill = getWaterfallEntryColor(
+      entry,
+      getWaterfallColors(settings?.colors, getColor),
+    );
+
+    return { x, y, width, height, fill };
+  };
+
+  // Used only for rendering data point values
+  const series = calculateWaterfallSeriesForValues(
+    data,
+    accessors,
+    settings?.showTotal,
+  );
+
+  return (
+    <svg width={layout.width} height={height}>
+      <Group top={topMargin} left={xMin}>
+        <GridRows
+          scale={yScale}
+          width={xMax}
+          strokeDasharray={layout.strokeDasharray}
+        />
+      </Group>
+
+      <AxisLeft
+        scale={yScale}
+        top={topMargin}
+        left={xMin}
+        label={leftLabel}
+        labelOffset={yLabelOffset}
+        hideTicks
+        hideAxisLine
+        labelProps={getLabelProps(chartStyle)}
+        tickFormat={value => formatNumber(value, settings?.y)}
+        tickLabelProps={() => getYTickLabelProps(chartStyle)}
+        {...axesProps}
+      />
+      <AxisBottom
+        scale={xScale}
+        left={xMin}
+        top={yMax + topMargin}
+        label={labels?.bottom}
+        tickLength={TICK_LENGTH}
+        numTicks={numTicks}
+        labelProps={merge(getLabelProps(chartStyle, getColor), {
+          dy:
+            -measureTextHeight(chartStyle.axes.labels.fontSize) +
```

Comment:
`-measureTextHeight(chartStyle.axes.labels.fontSize)` makes the label go back to almost at the same position the x-axis tick label is rendered, from here, we can then just add x-axis tick height and label padding to get the position the x-axis label is supposed to be rendered.